src/main/java/it/gov/pagopa/payhub/activities/service/treasury/TreasuryOpi14ValidatorService.java

```diff
@@ -0,0 +1,181 @@
+package it.gov.pagopa.payhub.activities.service.treasury;
+
+import it.gov.pagopa.payhub.activities.dto.treasury.TreasuryErrorDTO;
+import it.gov.pagopa.payhub.activities.util.TreasuryUtils;
+import it.gov.pagopa.payhub.activities.xsd.treasury.opi14.FlussoGiornaleDiCassa;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TreasuryOpi14ValidatorService implements TreasuryValidatorService<FlussoGiornaleDiCassa> {
+
+    private static final String NOT_AVAILABLE = "Not available";
+
+    List<TreasuryErrorDTO> treasuryErrorDTOList;
+
+    public TreasuryOpi14ValidatorService() {
+        treasuryErrorDTOList = new ArrayList<>();
```

Comment:
Is this class created for each row?
If not, you cannot use an instance variabile
If yes, probably you could avoid to create a new class for each record just to collect error data

Reply:
PS: you could also create n classes for each check, implementing the same interface
Next you could inject It using List<...> type in order to let Spring collect all the validators and execute the checks in a more readable and maintanable way (easier to test because you will have 1 class per check to test)